app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -271,7 +271,9 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
             // if handling new return code (want to return to home screen) but a return at the end of your statement
             switch(requestCode) {
             case PREFERENCES_ACTIVITY:
-                uiController.configUI();
+                // rebuild buttons in case language was changed
+                uiController.setupUI();
```

Comment:
Oh wow I was looking at the UI controller for LoginActivity, hence my confusion.